src/translator/batching_pool.cpp

```diff
@@ -49,8 +59,18 @@ size_t BatchingPool::enqueueRequest(Ptr<Request> request) {
     if (!request->cacheHitPrefilled(i)) {
       RequestSentence sentence(i, request);
       size_t bucket_id = sentence.numTokens();
-      assert(bucket_id < bucket_.size());
+
+      // Due to a workaround for pivoting, unless we can discipline the
```

Comment:
What is the motivation for changing the behaviour for non-pivoting requests?

`makeRequest` and `makePivotRequest` exist as separate methods, but not `enqueuePivotRequest`?

Also, do we need the resize in the constructor as well as here?

Reply:
Less duplication? The if is expected to be called very less?

Reply:
Ok, the idea being the single resize in the constructor with `pivotSlack` should avoid (most of) the incremental resizes.

Is there some noticeable impact on non-pivoting with this change?

Reply:
Seconds for WNGT20, 40 threads.

pivot | main
-- | --
321.8 ± 4.08σ | 322.1 ± 5.2σ